src/main/java/it/pagopa/pn/deliverypush/action/utils/CourtesyMessageUtils.java

```diff
@@ -242,17 +242,18 @@ private SendMessageRequestBody buildSendMessageRequest(NotificationInt notificat
                 .internalRecipientId(notification.getRecipients().get(recIndex).getInternalId())
                 .originId(notification.getIun())
                 .senderDescription(notification.getSender().getPaDenomination())
-                .associatedPayment(hasPagoPaPayment(notification));
+                .associatedPayment(hasRecipientPagoPaPayment(notification, recIndex));
     }
 
     private PnAuditLogEvent buildAuditLogEvent(String iun, int recIndex, String eventId) {
         return auditLogService.buildAuditLogEvent(iun, recIndex, PnAuditLogEventType.AUD_DA_SEND_TPP, "sendTppMessage eventId={}", eventId);
     }
 
-    private boolean hasPagoPaPayment(NotificationInt notification) {
-        return notification.getRecipients().stream()
-                .filter(recipient -> !CollectionUtils.isEmpty(recipient.getPayments()))
-                .flatMap(recipient -> recipient.getPayments().stream())
+    private boolean hasRecipientPagoPaPayment(NotificationInt notification, Integer recIndex) {
+        if(CollectionUtils.isEmpty(notification.getRecipients().get(recIndex).getPayments())) {
```

Comment:
@empassaro nel caso in cui ha pagamenti ma solo di tipo F24 questo controllo restituisce true invece dovremmo restituire true solo se abbiamo almeno un pagamento di tipo pagoPa